src/primitives/createFragment.ts

```diff
@@ -1,44 +1,94 @@
-import { getFragment } from "relay-runtime";
 import type { GraphQLTaggedNode } from "relay-runtime";
+import RelayRuntime from "relay-runtime/experimental";
 import type {
-	ArrayKeyType,
-	ArrayKeyTypeData,
 	KeyType,
 	KeyTypeData,
 } from "relay-runtime/lib/store/ResolverFragments";
-import { createResource } from "solid-js";
-import type { ResourceReturn } from "solid-js";
-import { unwrap } from "solid-js/store";
+import {
+	batch,
+	createComputed,
+	createMemo,
+	createResource,
+	onCleanup,
+} from "solid-js";
+import type { Accessor } from "solid-js";
 
+import { createStore, unwrap } from "solid-js/store";
 import { useRelayEnvironment } from "../RelayEnvironment";
-import { createFragmentNode } from "./createFragmentNode";
+import { type DataProxy, makeDataProxy } from "../utils/dataProxy";
+
+type FragmentResult<T> = {
+	data: T | undefined;
+	error: unknown;
+	pending: boolean;
+};
```

Comment:
I'm surprised you need this type. The [structure within Flow](https://github.com/facebook/relay/blob/9035b323ba799ae5971a99e4df0803a38b81955e/packages/relay-runtime/store/observeFragmentExperimental.js#L38) is a discriminated union which helps model the fact that only one of these states can ever be true at any one time. Either you have data, an error, or you are pending. Never more than one at a time. This type makes it harder to clarify that fact.